### PR TITLE
WIP [Active record]

### PR DESCRIPTION
**Pull request by githubjeka**
Problems `\humhub\components\ActiveRecord`:

- [ ] ActiveRecord's required `Yii::$app->user` component
- [ ] All ActiveRecord's models use FileManager
- [ ] Some core models used functional of ActiveRecord(`created_at`, `updated_by` ... )but haven't tables in database.
- [ ] One's of core models used only `updated_` fields, others - only `created_`
- [ ]   7 out of 13 cases `beforeSave()` is not involved for one reason or another. See test 
image

- [x] `getCreatedBy` and `getUpdatedBy` do not check the existence of a property `created_by` and `updated_by`

### CODE REVIEW

**Comment by githubjeka** [comment]
I propose to add `@deprecated` to current AR and move functional AR to specific places, for example, traits

**Comment by luke-** [comment]
@githubjeka Good point! Should be no problem to mark the HumHub AR as deprecated and implement the functions as Traits. From our side you could start - thank you!

**Comment by luke-** [comment]
There are also some Events using the HumHub AR.
https://github.com/humhub/humhub/blob/master/protected/humhub/modules/notification/config.php#L21

We also need to change this to the default Yii2 AR.

**Comment by danielkesselberg** [comment]
We may consider http://www.yiiframework.com/doc-2.0/yii-behaviors-timestampbehavior.html and http://www.yiiframework.com/doc-2.0/yii-behaviors-blameablebehavior.html.

**Comment by githubjeka** [comment]
So... Need to think more about behaviors and theirs expand

**Comment by githubjeka** [comment]
Added traits. Need your review and thoughts

File: protected/humhub/modules/user/models/Invite.php [hunk 1]
```diff
@@ -21,16 +20,13 @@
...
- * @property string $created_at
- * @property integer $created_by
- * @property string $updated_at
- * @property integer $updated_by
  * @property string $language
  * @property string $firstname
  * @property string $lastname
  */
-class Invite extends ActiveRecord
+class Invite extends \humhub\db\HActiveRecord
```
Comment (githubjeka) [inline protected/humhub/modules/user/models/Invite.php]: Removed `FileManager` functional for Invite model.
Meta information about created_..., updated_... available by `\humhub\db\traits\MetaActiveRecordTrait`

File: protected/humhub/modules/user/models/Invite.php [hunk 2]
```diff
@@ -90,7 +86,7 @@ public function beforeSave($insert)
         if ($insert && $this->token == '') {
             $this->token = Yii::$app->security->generateRandomString(self::TOKEN_LENGTH);
         }
-
+        $this->setMeta($insert, $this->user_originator_id);
```
Comment (githubjeka) [inline protected/humhub/modules/user/models/Invite.php]: Need to throw false in beforeSave() If created_by or updated_by will set to NULL(user_id is wrong) then?

Comment (githubjeka) [reply protected/humhub/modules/user/models/Invite.php]: For example:

```php
public function beforeSave($insert)
    {
        if (!parent::beforeSave($insert)) {
            return false;
        }
        if ($insert && $this->token == '') {
            $this->token = Yii::$app->security->generateRandomString(self::TOKEN_LENGTH);
        }
        return $this->setMeta($insert, $this->user_originator_id);
    }
```

**Comment by buddh4** [comment]
As a small note I'am not a fan of those prefixed class names like`HActiveRecord`. I Think `humhub\db\ActiveRecord` should be enought to distinguish the classes, we do the same for other components as controllers etc.

**Pull request closed without merging**